tools/bridge/bridge/transfer_fetcher.py

```diff
@@ -1,25 +1,135 @@
 import logging
-
 import gevent
-from gevent.queue import Queue
 
+from gevent.queue import Queue
 from web3 import Web3
 
+MINIMAL_TOKEN_ABI = [
+    {
+        "anonymous": False,
+        "inputs": [
+            {"indexed": True, "name": "from", "type": "address"},
+            {"indexed": True, "name": "to", "type": "address"},
+            {"indexed": False, "name": "value", "type": "uint256"},
+        ],
+        "name": "Transfer",
+        "type": "event",
+    }
+]
+
+TRANSFER_EVENT_SIGNATURE_HASH = Web3.keccak(text="Transfer(address,address,uint256)")
+
+
+class TransferFetcher:
+    def __init__(
+        self,
+        *,
+        queue: Queue,
+        w3_foreign: Web3,
+        token_contract_address: str,
```

Comment:
Suggestion: Use the web3 contract object instead as it's provides a easy to use ABI, assuming it's possible to use `getLogs` instead of `getFilterLogs` with that object.

Reply:
I'm confused. I'm actually using the contract. But I don't want the main module to do more than providing me the configured address. So I don't see what to do here. Anyways I probably need to switch, since in the below comments we figured out that this will not use `getLogs`.